src/libfuncs/enum.rs

```diff
@@ -94,29 +95,10 @@ pub fn build_init<'ctx, 'this>(
                 false,
             );
 
-            let tag_val = entry
-                .append_operation(arith::constant(
-                    context,
-                    IntegerAttribute::new(tag_ty, info.index as i64).into(),
-                    location,
-                ))
-                .result(0)?
-                .into();
+            let tag_val = entry.const_int(context, location, info.index as i64, 64)?;
```

Comment:
the cast is not needed here, also you should use the tag_ty, so instead of const_int use const_int_from_type and pass tag_ty along the info.index